examples/nvme/perf/perf.c

```diff
@@ -632,27 +632,42 @@ static const struct ns_fn_table nvme_fn_table = {
 	.cleanup_ns_worker_ctx	= nvme_cleanup_ns_worker_ctx,
 };
 
-static void
+static int
 build_nvme_name(char *name, size_t length, struct spdk_nvme_ctrlr *ctrlr)
 {
 	const struct spdk_nvme_transport_id *trid;
+	int res = 0;
 
 	trid = spdk_nvme_ctrlr_get_transport_id(ctrlr);
 
 	switch (trid->trtype) {
 	case SPDK_NVME_TRANSPORT_PCIE:
-		snprintf(name, length, "PCIE (%s)", trid->traddr);
+		res = snprintf(name, length, "PCIE (%s)", trid->traddr);
 		break;
 	case SPDK_NVME_TRANSPORT_RDMA:
-		snprintf(name, length, "RDMA (addr:%s subnqn:%s)", trid->traddr, trid->subnqn);
+		res = snprintf(name, length, "RDMA (addr:%s subnqn:%s)", trid->traddr, trid->subnqn);
 		break;
 	case SPDK_NVME_TRANSPORT_TCP:
-		snprintf(name, length, "TCP  (addr:%s subnqn:%s)", trid->traddr, trid->subnqn);
+		res = snprintf(name, length, "TCP  (addr:%s subnqn:%s)", trid->traddr, trid->subnqn);
 		break;
+
 	default:
 		fprintf(stderr, "Unknown transport type %d\n", trid->trtype);
 		break;
 	}
+	return res;
+}
+
+static void
+build_nvme_ns_name(char *name, size_t length, struct spdk_nvme_ctrlr *ctrlr, uint32_t nsid)
+{
+	int res = 0;
+
+	res = build_nvme_name(name, length, ctrlr);
+	if (res > 0) {
+		snprintf(name + res - 1, length - res + 1, " NSID %d", nsid);
```

Comment:
As far as I know, snprintf returns the number of char written without termination symbol, so why do we need these `+1` and `-1`? I might be wrong, but it seems that we need just `name + res, length - res`
Also please correct format of nsid, it should be %u